PureLayout/PureLayout/ALView+PureLayout.m

```diff
@@ -172,13 +179,13 @@ - (NSLayoutConstraint *)autoPinEdgeToSuperviewEdge:(ALEdge)edge withInset:(CGFlo
     ALView *superview = self.superview;
     NSAssert(superview, @"View's superview must not be nil.\nView: %@", self);
     if (edge == ALEdgeBottom || edge == ALEdgeRight || edge == ALEdgeTrailing) {
-        // The bottom, right, and trailing insets (and relations, if an inequality) are inverted to become offsets
-        inset = -inset;
-        if (relation == NSLayoutRelationLessThanOrEqual) {
-            relation = NSLayoutRelationGreaterThanOrEqual;
-        } else if (relation == NSLayoutRelationGreaterThanOrEqual) {
-            relation = NSLayoutRelationLessThanOrEqual;
-        }
+        // This must be formulated a bit differently, with the superview first, in order to keep the constant
+        // positive.
+        self.translatesAutoresizingMaskIntoConstraints = NO;
+        NSLayoutAttribute edgeAttribute = [NSLayoutConstraint al_layoutAttributeForAttribute:(ALAttribute)edge];
+        NSLayoutConstraint *constraint = [NSLayoutConstraint constraintWithItem:superview attribute:edgeAttribute relatedBy:relation toItem:self attribute:edgeAttribute multiplier:1.0 constant:inset];
+        [constraint autoInstall];
+        return constraint;
```

Comment:
The original implementation is cleaner, but it is a leaky abstraction. Users that want to adjust the constant on the constraint need to know when PureLayout decided to make the constraint negative or not. Given that the user passed in a positive constant it seems wrong to make that negative.

That said, users of PureLayout may already have workarounds for using negative constants in certain cases. If we fix this bug it might trip them up.

Also, keeping the constant positive makes debugging easier. The constraint prints to the debugger in an easier format (I wrote that code when I worked on Auto Layout at Apple. Sorry I didn't just detect this other case...)

<img width="1168" alt="screen shot 2016-01-10 at 10 21 52 am" src="https://cloud.githubusercontent.com/assets/1337793/12222235/7d888d94-b784-11e5-926b-7046fca80cb3.png">